### DISCUSSION TITLE
Generate static images from server images

### DISCUSSION THREAD

**Discussion by beschler** [vercel/next.js#62923] [Help]
### Summary

(I'm using Next.js v14.1.0, and the App Router.)

I'm building a blog-type website using [static exports](https://nextjs.org/docs/app/building-your-application/deploying/static-exports) (formerly [getStaticProps](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props) and [getStaticPaths](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-paths)) to create a static site, and I'm connecting to an AWS S3 bucket where my images for my website content are stored, and referenced in a headless CMS. (For example, certain images from S3 are attached to certain pages of content.)

I know Next.js can create static paths for dynamic routes with the content retrieved from an API (e.g. a headless CMS). But is it possible to have Next.js download and convert my images from S3 into static images in (for example) the `/public` folder during the build process, so that the [`<Image>` components](https://nextjs.org/docs/pages/api-reference/components/image) across my site will reference these local assets instead of the absolute S3 URLs in the production build?

Thanks so much!

### Additional information

```js
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.6.0: Tue Nov  7 21:42:24 PST 2023; root:xnu-8796.141.3.702.9~2/RELEASE_ARM64_T6020
Binaries:
  Node: 19.8.1
  npm: 9.5.1
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.0
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: N/A
Next.js Config:
  output: N/A
```

**Comment by chenzhutian** [comment]
I have the same issue of using Notion as the CMS
